pkg/disttask/framework/taskexecutor/manager.go

```diff
@@ -410,6 +434,14 @@ func (m *Manager) handleExecutableTask(task *proto.Task) {
 	}
 }
 
+func (m *Manager) getStepResource(concurrency int) *proto.StepResource {
+	return &proto.StepResource{
+		CPU: proto.NewAllocatable(int64(concurrency)),
+		// same proportion as CPU
+		Mem: proto.NewAllocatable(int64(float64(concurrency) / float64(m.totalCPU) * float64(m.totalMem))),
```

Comment:
assert concurrency < totalCPU?

Reply:
it will fail to alloc slots, shouldn't reach here

Reply:
Yep, I mean just add some comments